artifact/tests/test_artifact.py

```diff
@@ -67,6 +67,42 @@ def test_dirs(self):
         self.assertTrue(exists(self.artifact.cwd))
         self.assertTrue(exists(self.artifact.path))
 
+class TestArtifactSimilarity(unittest.TestCase):
+
+    def setUp(self):
+        self.artifactA = artifact.Artifact({
+            '_id': uuid4(),
+            'name': 'artifact-A',
+            'type': 'type-A',
+            'documentation': "This is a description of artifact A",
+            'command': ['ls', '-l'],
+            'path': '/',
+            'hash': hashlib.md5().hexdigest(),
+            'git': artifact.artifact.getGit('.'),
+            'cwd': '/',
+            'inputs': [],
+        })
+
+        self.artifactB = artifact.Artifact({
+            '_id': uuid4(),
+            'name': 'artifact-B',
+            'type': 'type-B',
+            'documentation': "This is a description of artifact B",
+            'command': ['ls', '-l'],
+            'path': '/',
+            'hash': hashlib.md5().hexdigest(),
+            'git': artifact.artifact.getGit('.'),
+            'cwd': '/',
+            'inputs': [],
+        })
+
+        self.artifactC = self.artifactA
```

Comment:
I don't think this is a deep copy, so it will always be the same (the eq test below isn't testing anything).

I would make two more, one that was *exactly* the same as A and one that had the same id and hash, but the rest differed to test similarity and equality.